lib/core/src/server/types.ts

```diff
@@ -0,0 +1,120 @@
+import { Configuration, Stats } from 'webpack';
+import { TransformOptions } from '@babel/core';
+import { typeScriptDefaults } from './config/defaults';
+
+/**
+ * ⚠️ This file contains internal WIP types they MUST NOT be exported outside this package for now!
+ */
+export interface ManagerWebpackOptions {
+  configDir: any;
+  configType?: string;
+  docsMode?: boolean;
+  entries: string[];
+  refs: Record<string, Ref>;
+  uiDll: boolean;
+  dll: boolean;
+  outputDir?: string;
+  cache: boolean;
+  previewUrl?: string;
+  versionCheck: VersionCheck;
+  releaseNotesData: ReleaseNotesData;
+  presets: any;
+}
+
+export interface Presets {
+  apply(
```

Comment:
I see the point but isn't too limitative (and potentially incomplete) to list all presets ?

Reply:
I somehow need these types to avoid type inference to "return" `any` on all these calls: https://github.com/storybookjs/storybook/pull/12839/files#diff-d0f0c842d74746ce0b792a604e01fb7ded59c01898321fb1b482ba3ca016ab7dR12-R28

It's not directly the presets but the functions a preset can expose and so for instance `webpack` function should/must always return a webpack `Configuration` whatever the preset. Do you agree? 🤔 

But it's not 100% clear for me yet, maybe I will get rid of that when improving the typing.

I'm surely missing some parameters that why I introduced a fallback on [L45](https://github.com/storybookjs/storybook/pull/12839/files#diff-672824f9073da83026877803f53d07dc680263367ef9da36b14a016c72bcf2bbR45)